openapi.yaml

```diff
@@ -0,0 +1,712 @@
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service cannot be deleted)_
+      security:
+        - subscription_key: []
+  /services/{serviceId}/logo:
+    put:
+      tags:
+        - services
+      summary: Upload service logo
+      description: |
+        Upload service logo by by `serviceId`. <br><br>_Note: This API operation needs the __Manage API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
+      operationId: updateServiceLogo
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Service logo payload _(base64 string representation)_
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Logo'
+        required: true
+      responses:
+        '200':
+          description: Successful operation _(service logo updated)_
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service logo cannot be updated)_
+      security:
+        - subscription_key: []
+  /services/{serviceId}/key:
+    put:
+      tags:
+        - services-authorization
+      summary: Regenerate service key
+      description: |
+        Regenerate service key by `serviceId`. <br><br>_Note: This API operation needs the __Manage API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
+      operationId: regenerateServiceKey
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Service key type
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SubscriptionKeyTypePayload'
+        required: true
+      responses:
+        '200':
+          description: Successful operation _(service key regenerated)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SubscriptionKeys'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service key cannot be regenerated)_
```

Comment:
Why "_service key cannot be regenerated_"?

Reply:
Same as previous similar comment